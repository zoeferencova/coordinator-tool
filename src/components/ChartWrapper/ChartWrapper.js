import React, { Component } from 'react';
import TimespanChart from '../TimespanChart/TimespanChart';

import styles from './ChartWrapper.module.css'

export default class ChartWrapper extends Component {
    componentDidMount() {
        this.setState({ chart: new TimespanChart(this.refs.chart) })
    }

    shouldComponentUpdate() {
        return false;
    }

    componentWillReceiveProps(nextProps) {
        this.state.chart.update(nextProps.dataType)
    }

    render() {
        return (
            <div ref="chart"></div>
        );
    }
}
