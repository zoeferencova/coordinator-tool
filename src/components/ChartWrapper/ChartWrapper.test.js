import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ChartWrapper from './ChartWrapper'

describe(`Chart Wrapper component`, () => {
  it('renders a .ChartWrapper by default', () => {
    const wrapper = shallow(<ChartWrapper />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})