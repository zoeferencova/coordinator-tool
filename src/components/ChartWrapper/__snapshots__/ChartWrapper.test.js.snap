// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Wrapper component renders a .ChartWrapper by default 1`] = `<div />`;
